### ISSUE TITLE
Error when reading OP_RETURN data unrelated to POET

### ISSUE THREAD

**Issue by lautarodragan** [poetapp/node#551]
https://github.com/poetapp/node/blob/8e19bce6d2cbecebe3b2ab465c46f4d933d59660/src/BlockchainReader/Bitcoin.ts#L45-L57

When reading data from the blockchain, nothing's validating that the length of the data stored in the OP_RETURN output.

This leads to `dataToPoetAnchor` trying to read bytes outside the range of the buffer, which throws:

```
    method: "scanNewBlock"
    error: {
      "stack": [
        "RangeError [ERR_OUT_OF_RANGE]: The value of \"offset\" is out of range. It must be >= 0 and <= 4. Received 6",
        "    at boundsError (internal/buffer.js:53:9)",
        "    at Buffer.readInt8 (internal/buffer.js:309:5)",
        "    at Object.exports.dataToPoetAnchor.data [as dataToPoetAnchor] (/usr/src/app/dist/babel/src/BlockchainReader/Bitcoin.js:27:36)",
        "    at dataOutputToPoetTransactionAnchor (/usr/src/app/dist/babel/src/BlockchainReader/Bitcoin.js:16:28)",
        "    at transactionToPoetAnchor (/usr/src/app/dist/babel/src/BlockchainReader/Bitcoin.js:11:26)",
        "    at Array.map (<anonymous>)",
        "    at Object.exports.blockToPoetAnchors.block [as blockToPoetAnchors] (/usr/src/app/dist/babel/src/BlockchainReader/Bitcoin.js:7:48)",
        "    at ClaimController.scanBlock (/usr/src/app/dist/babel/src/BlockchainReader/ClaimController.js:46:35)",
        "    at process._tickCallback (internal/process/next_tick.js:68:7)"
      ],
      "message": "The value of \"offset\" is out of range. It must be >= 0 and <= 4. Received 6",
      "code": "ERR_OUT_OF_RANGE",
      "type": "NodeError"
    }
    blockHeight: 1280418
    lastBlockHeight: 1280417
```

We could solve this by having `dataToPoetAnchor` return a `undefined`. 

```ts
export const dataToPoetAnchor = (data: string): PoetAnchor => {
  const buffer = Buffer.from(data, 'hex')
  if (buffer.byteLength < 4 + 2 + 1 + N)
    return undefined
  const prefix = buffer.slice(0, 4).toString()
  const version = Array.from(buffer.slice(4, 6))
  const storageProtocol = buffer.readInt8(6)
  const ipfsDirectoryHash = buffer.slice(7).toString()
  return {
    storageProtocol,
    prefix,
    version,
    ipfsDirectoryHash,
  }
}
```

Where `N` is a dynamic value that depends on the storageProtocol and multihash. We're only supporting one version of multihash and storageProtocol at the moment, so it's safe to hard-code it, but we'll want to use a constant rather than a magic number.

__Note__

Returning `Option<PoetAnchor>` would be nicer but we don't have anything to support Option in the code base yet, and it'd require a bit of refactor.

Returning undefined will solve the issue immediately since `blockToPoetAnchors` has a `.filter(isTruthy)`.

__FYI__

The block is [1280418](https://test-insight.bitpay.com/api/block/00000000000012ef158b07cb514f14156cf99a628eff53f789db05af1a1bbd88).

It has three transactions with data:
- [81120518637a0b964da9f9075995335eb12397100e5f7f9a6e711c16b5f292f6](https://test-insight.bitpay.com/api/tx/81120518637a0b964da9f9075995335eb12397100e5f7f9a6e711c16b5f292f6)
- [918cc48177c39a01ab6a20d3e980526775262996de00928fa92d163bd2508812](https://test-insight.bitpay.com/api/tx/918cc48177c39a01ab6a20d3e980526775262996de00928fa92d163bd2508812)
- [7fa11a4809441799f6392fe06c1ba13dca53e72b5524f5be23d4560262cc9b67](https://test-insight.bitpay.com/api/tx/7fa11a4809441799f6392fe06c1ba13dca53e72b5524f5be23d4560262cc9b67)

None of those is a Po.et anchor.

`918cc48177c39a01ab6a20d3e980526775262996de00928fa92d163bd2508812` is probably the one throwing the stack trace shown above:

```js
> Buffer.from('1437681861', 'hex')
<Buffer 14 37 68 18 61>
```

**Comment by geoffturk** [comment]
@jwicks31 Please fix ASAP. Thanks.

**Comment by geoffturk** [comment]
FYI, I noticed that the our node dbs had stopped writing data into the blockchainReader collection after 867 blocks. 1280418 - 1279551 = 867, so they definitely got stuck on the block you identified.

We deployed the code from Jesse's PR onto one of our node instances in QA, and it has now scanned past that block and is still incrementing. 👍 

![image](https://user-images.githubusercontent.com/32190662/47142062-5e287c00-d2c2-11e8-935d-19506be0f13d.png)

@WesleyCharlesBlake FYI
